fuzzers/libfuzzer_interceptors/fuzzer.py

```diff
@@ -0,0 +1,86 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Integration code for libFuzzer fuzzer."""
+
+import subprocess
+import os
+
+from fuzzers import utils
+
+
+def build():
+    """Build benchmark."""
+    # With LibFuzzer we use -fsanitize=fuzzer-no-link for build CFLAGS and then
+    # /usr/lib/libFuzzer.a as the FUZZER_LIB for the main fuzzing binary. This
+    # allows us to link against a version of LibFuzzer that we specify.
+    cflags = ['-fsanitize=fuzzer-no-link']
+
+    # Can be removed once the patch https://reviews.llvm.org/D83987
+    # appears in gcr.io/fuzzbench/base-builder
+    cflags += ['-fno-builtin-bcmp']
+    cflags += ['-fno-builtin-memcmp']
+    cflags += ['-fno-builtin-strncmp']
+    cflags += ['-fno-builtin-strcmp']
+    cflags += ['-fno-builtin-strncasecmp']
+    cflags += ['-fno-builtin-strcasecmp']
+    cflags += ['-fno-builtin-strstr']
+    cflags += ['-fno-builtin-strcasestr']
+    cflags += ['-fno-builtin-memmem']
+
+    utils.append_flags('CFLAGS', cflags)
+    utils.append_flags('CXXFLAGS', cflags)
+
+    os.environ['CC'] = 'clang'
+    os.environ['CXX'] = 'clang++'
+    os.environ['FUZZER_LIB'] = '/usr/lib/libFuzzer.a'
+
+    utils.build_benchmark()
+
+
+def fuzz(input_corpus, output_corpus, target_binary):
+    """Run fuzzer. Wrapper that uses the defaults when calling
+    run_fuzzer."""
+    run_fuzzer(input_corpus, output_corpus, target_binary)
+
+
+def run_fuzzer(input_corpus, output_corpus, target_binary, extra_flags=None):
+    """Run fuzzer."""
+    if extra_flags is None:
+        extra_flags = []
+
+    flags = [
+        '-print_final_stats=1',
+        # `close_fd_mask` to prevent too much logging output from the target.
+        '-close_fd_mask=3',
+        # Run in fork mode to allow ignoring ooms, timeouts, crashes and
+        # continue fuzzing indefinitely.
+        '-fork=1',
+        '-ignore_ooms=1',
+        '-ignore_timeouts=1',
+        '-ignore_crashes=1',
+
```

Comment:
Not relevant to this PR in particular but there is no timeout here since I prefer defaults are used. But libFuzzer's default here probably hurts it (1-hour) I think libFuzzer's default timeout should be lowered. Especially for fork mode where it doesn't mean the end of fuzzing. I assume the generous timeout was a result of libFuzzer's single threaded days.